com.unity.multiplayer.mlapi/Runtime/Exceptions/SpawnStateException.cs

```diff
@@ -25,9 +25,4 @@ public class SpawnStateException : Exception
         /// <param name="inner">The inner exception</param>
         public SpawnStateException(string message, Exception inner) : base(message, inner) { }
     }
-
-    public class InvalidChannelException : Exception
```

Comment:
Removing this because we shouldn't have specific exceptions like this. This should have just been an assert.

Reply:
Why shouldn't we have explicit expectations?

Reply:
Oh my comment wasn't very detailed, let me try to explain. I think we should have explicit exceptions. A great example for this would be the NotServerException which we have in MLAPI.

The only use case of this InvalidChannelException was throwing when trying to add a user channel to an existing channel dictionary. I don't see why we need a special operation type for this as the ArgumentException of dictionary add or InvalidOperationException already cover this use case.

If we introduce new exceptions we should do it properly. Meaning put them in a separate file, add documentation explaining when this type of exception is thrown etc. InvalidChannelException sounds to me like something which would be thrown when trying to send data on a not supported channel for the given send operation. But that's not how this exception was used in code.

I also think the dotnet exceptions best practices guide is a great starting point for how exception handling should be done. There's a [section](https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions#use-the-predefined-net-exception-types) about using existing exceptions where they apply.

Reply:
Makes sense thanks for the clarification.

Reply:
+1

Reply:
but again, more than dictionary lookups changes :P

Reply:
I have removed  this change from this PR. Will create a separate PR later to clean up the exception